### ISSUE TITLE
Resuming collection after interruption

### ISSUE THREAD

**Issue by igorbrigadir** [DocNow/twarc#589]
It should be possible to pick up where you left off, for a long running search - if you read the last request and use the pagination token. As far as i can tell, these pagination tokens don't expire. Will try and see how long they last (in case they do). This could also be used to resume searches across months - if you run out of your monthly quota or something.

Ideally it will be a `--resume` option - with a `@resumable` decorator on the command line options that can read an existing file, and continue appending and paginating given an existing pagination token. The client code should already support this, so these changes are mostly in the command line tool.

**Comment by Michael-Gauthier** [comment]
Hello @igorbrigadir, and everyone else! This is pretty crazy, because I came here to post the exact same question because of disconnection problems I randomly have in my office... So I do not really understand: is it a feature you are planning to add, or is it already possible to resume collections after interruptions? Thanks a lot in advance, as usual! : )

**Comment by edsu** [comment]
I agree it would be a nice feature, and shouldn't be too tricky given `next_token` is persisted to the output?

**Comment by Michael-Gauthier** [comment]
Gods, when it is implemented, I'll send you guys chocolates or whatever, that will be so useful! ^^

Thanks again for your hard work and efforts to keep improving the tool by the way! : )

**Comment by igorbrigadir** [comment]
> is it a feature you are planning to add, or is it already possible to resume collections after interruptions?

Kinda both, it doesn't exist in the command line twarc yet, but it's possible to do this with the library, https://twittercommunity.com/t/pulling-a-large-data-set-with-twarc2-client/165685/6?u=igorbrigadir all you need is to extract a `next_token` and specify it as a parameter into the function and it should continue - but i need to double check that, and make sure that using old next_token values actually works like this.

**Comment by Michael-Gauthier** [comment]
Thanks for the feedback, and for the link to the thread! So, if I understand correctly, I would need to find the <next_token>, add this to my "normal" twarc2 query program, and it should resume where it stopped?

**Comment by igorbrigadir** [comment]
Yes, except there's no way to add the `next_token` to `twarc2` yet - i'll have add that part in.

**Comment by Michael-Gauthier** [comment]
Ok, thanks a lot for your reactivity and your clarification! : )
